src/conduits/java/com/enderio/conduits/common/network/ConduitSavedData.java

```diff
@@ -168,6 +195,31 @@ private void merge(GraphObject<Mergeable.Dummy> object, List<Pair<GraphObject<Me
         }
     }
 
+    public NodeIdentifier takeNodeIdentifier(IConduitType type, BlockPos pos) {
+        Map<ChunkPos, Map<BlockPos, NodeIdentifier>> typeMap = deserializedNodes.get(type);
+        if (typeMap == null) throw new IllegalStateException("Conduit data is missing!");
+
+        ChunkPos chunkPos = new ChunkPos(pos);
+        Map<BlockPos, NodeIdentifier> chunkMap = typeMap.get(new ChunkPos(pos));
+        if (chunkMap == null) throw new IllegalStateException("Conduit data is missing!");
+
+        NodeIdentifier node = chunkMap.get(pos);
+        if (node == null) throw new IllegalStateException("Conduit data is missing!");
```

Comment:
use Objects#requireNonNull`
```suggestion
        Map<ChunkPos, Map<BlockPos, NodeIdentifier>> typeMap = Objects.requireNonNUll(deserializedNodes.get(type), "Conduit data is missing!");
        ChunkPos chunkPos = new ChunkPos(pos);
        Map<BlockPos, NodeIdentifier> chunkMap = Objects.requireNonNull(typeMap.get(new ChunkPos(pos), "Conduit data is missing!");
        NodeIdentifier node = Objects.requireNonNull(chunkMap.get(pos), "Conduit data is missing!");
```